src/primitives/transaction_struct.zig

```diff
@@ -0,0 +1,46 @@
+const std = @import("std");
+
+const scriptWitness = struct {
+    stack: [][]u8,
+};
+
+/// A combination of a transaction hash and an index n into its vout
+const OutPoint = struct {
+    hash: u32, 
+    index: u32,
+};
+
+// An input of a transaction.  It contains the location of the previous
+// transaction's output that it claims and a signature that matches the
+// output's public key.
+const TxIn = struct {
+    prevOutPoint: OutPoint,
+    sigScript: []u8,
+    sequence: u32,
+    txWitness: scriptWitness,
+};
+
+const TxOut = struct {
+    value: i64,
+    scriptPubKey: []u8,
+};
+
+const Transaction = struct {
+    vin: []TxIn,
+    vout: []TxOut,
+    version: u32,
+    lockTime: u32,
+
+    pub fn init(vin: []TxIn, vout: []TxOut, version: u32, lockTime: u32) Transaction {
```

Comment:
the name init might be confusing, usually init in Zig is used when you need to allocate memory, to follow the init / deinit pattern. Here there is no memory allocation, and hence also no deinit function